### ISSUE TITLE
Querying Multi-Channel Funnels Reporting API

### ISSUE THREAD

**Issue by Atlasp** [panalysis/Google2Pandas#41]
I've been using : 

```
query = {
        'reportRequests': [
        {
          'viewId': 'viewid',
          'dateRanges': [{'startDate': '2018-02-01', 'endDate': '2018-02-28'}],
          'metrics': [{'expression': 'ga:adClicks'},
                      {'expression': 'ga:adCost'},
                      {'expression': 'ga:CPC'},
                      {'expression': 'ga:sessions'},
                      {'expression': 'ga:bounceRate'},
                      {'expression': 'ga:pageviewsPerSession'},
                      {'expression': 'ga:goal6ConversionRate'},
                      {'expression': 'ga:goal6Completions'},
                      {'expression': 'ga:goal6Value'}],
          'dimensions': [{'name': 'ga:campaign'},
                         {'name': 'ga:date'}]
        }]
      }


conn = GoogleAnalyticsQueryV4(secrets='secret.json')
df = conn.execute_query(query)

print(df)
```

with success, however, I am unable to query data from Multi-Channel Funnels Reporting API, are there any examples how to modify the query so mcf dimensions and metrics would be usable?

**Comment by DeliciousHair** [comment]
I have not been using any of these tools (GA) at all recently, and as a result I don't have a quick-fix answer that I can promise will work.

However, the V4 api is much simpler than the V3, and `GoogleAnalyticsQueryV4` is basically a convenience wrapper for `googleapiclient.discovery.build`, which [requires a serviceName argument](https://google.github.io/google-api-python-client/docs/epy/googleapiclient.discovery-module.html#build). At the moment, I've hard-coded in a value of `analytics`, so if you simply make this value a keyword argument to `OAuthDataReaderV4._init_service()`, then you could subclass out any of the api services like I've done with `GoogleAnalyticsQueryV4` in a rather trivial manner.

I won't have time to try this myself for at least a few days at this point; if you work something out please do submit a pr--any and all help appreciated!

Thanks!

**Comment by rahulpanchal0002** [comment]
I am too facing the same problem. I tried to write the query different ways, but still produces error
ValueError: The (required) 'start_date' parameter is missing or invalid

I have start_date in my query.
Is there any example for building MCF query?

**Comment by Archirk** [comment]
Oh.. two years have passed and answer is still desired.

**Comment by DeliciousHair** [comment]
Indeed! And now I may be using this again, so I'll try to get to this later in the week. Or so.
